### PR TITLE
Change all MasterRequest calls to MainRequest

### PR DESCRIPTION
**Pull request by Roshyo**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | removing deprecations
| Related tickets | 
| License         | MIT

For now, it should accept both until Symfony <5.3 support is dropped

### CODE REVIEW

**Review by vvasiloi — APPROVED** [review]

**Comment by vvasiloi** [comment]
I was almost sure this was already merged...

**Comment by Roshyo** [comment]
Yeah, me too :thinking: IDK what happened

**Comment by vvasiloi** [comment]
Looks like the it didn't re-run the jobs...
Can you rebase and push, @Roshyo?

**Review by loic425 — APPROVED** [review]

**Review by stloyd — APPROVED** [review]

**Comment by 4c0n** [comment]
So in the end there is no mention of having to remove the `if` statements at some point in the future anywhere?
I don't want to block merging this PR, but I do feel we need at least something as a reminder that this requires work when `isMasterRequest()` is finally fased out.

Could be a github issue or PR as well as far as I'm concerned.

**Comment by Roshyo** [comment]
I'm ok to write it somewhere, but I can't find any md file that would fit. Perhaps an issue is the way to go :thinking:

File: src/Sylius/Bundle/AdminBundle/EventListener/ResourceDeleteSubscriber.php [hunk 1]
```diff
@@ -50,7 +50,12 @@ public function onResourceDelete(ExceptionEvent $event): void
             return;
         }
 
-        if (!$event->isMasterRequest() || 'html' !== $event->getRequest()->getRequestFormat()) {
+        if (\method_exists($event, 'isMainRequest')) {
+            $isMainRequest = $event->isMainRequest();
+        } else {
+            $isMainRequest = $event->isMasterRequest();
```
[the remark this answers is not in the document]
Comment (lchrusciel) [reply src/Sylius/Bundle/AdminBundle/EventListener/ResourceDeleteSubscriber.php:56]: I think, we can add note to: https://github.com/Sylius/Sylius/issues/13274

File: src/Sylius/Bundle/ApiBundle/Context/TokenValueBasedCartContext.php [hunk 2]
```diff
@@ -57,20 +57,24 @@ public function getCart(): OrderInterface
         return $cart;
     }
 
-    private function getMasterRequest(): Request
+    private function getMainRequest(): Request
     {
-        $masterRequest = $this->requestStack->getMasterRequest();
-        if (null === $masterRequest) {
-            throw new CartNotFoundException('There is no master request on request stack.');
+        if (\method_exists($this->requestStack, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Context/TokenValueBasedCartContext.php:62]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists($this->requestStack, 'getMainRequest')) {
```

File: src/Sylius/Bundle/ApiBundle/spec/Context/TokenValueBasedCartContextSpec.php [hunk 3]
```diff
@@ -43,18 +43,26 @@ function it_returns_cart_by_token_value(
         $request->attributes = new ParameterBag(['tokenValue' => 'TOKEN_VALUE']);
         $request->getRequestUri()->willReturn('/api/v2/orders/TOKEN_VALUE');
 
-        $requestStack->getMasterRequest()->willReturn($request);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/spec/Context/TokenValueBasedCartContextSpec.php:46]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Bundle/ApiBundle/spec/Context/TokenValueBasedCartContextSpec.php [hunk 4]
```diff
@@ -43,18 +43,26 @@ function it_returns_cart_by_token_value(
...
+        }
         $orderRepository->findCartByTokenValue('TOKEN_VALUE')->willReturn($cart);
 
         $this->getCart()->shouldReturn($cart);
     }
 
     function it_throws_an_exception_if_there_is_no_master_request_on_request_stack(RequestStack $requestStack): void
     {
-        $requestStack->getMasterRequest()->willReturn(null);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/spec/Context/TokenValueBasedCartContextSpec.php:58]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Bundle/ApiBundle/spec/Context/TokenValueBasedCartContextSpec.php [hunk 5]
```diff
@@ -81,7 +93,11 @@ function it_throws_an_exception_if_there_is_no_token_value(
         $request->attributes = new ParameterBag([]);
         $request->getRequestUri()->willReturn('/api/v2/orders');
 
-        $requestStack->getMasterRequest()->willReturn($request);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/spec/Context/TokenValueBasedCartContextSpec.php:96]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Bundle/ApiBundle/spec/Context/TokenValueBasedCartContextSpec.php [hunk 6]
```diff
@@ -66,10 +74,14 @@ function it_throws_an_exception_if_the_request_is_not_an_api_request(
         $request->attributes = new ParameterBag([]);
         $request->getRequestUri()->willReturn('/orders');
 
-        $requestStack->getMasterRequest()->willReturn($request);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/spec/Context/TokenValueBasedCartContextSpec.php:77]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Bundle/ApiBundle/spec/Context/TokenValueBasedCartContextSpec.php [hunk 7]
```diff
@@ -97,7 +113,11 @@ function it_throws_an_exception_if_there_is_no_cart_with_given_token_value(
         $request->attributes = new ParameterBag(['tokenValue' => 'TOKEN_VALUE']);
         $request->getRequestUri()->willReturn('/api/v2/orders/TOKEN_VALUE');
 
-        $requestStack->getMasterRequest()->willReturn($request);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/spec/Context/TokenValueBasedCartContextSpec.php:116]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Bundle/ChannelBundle/Context/FakeChannel/FakeChannelContext.php [hunk 8]
```diff
@@ -58,13 +58,17 @@ public function getChannel(): ChannelInterface
     /**
      * @throws ChannelNotFoundException
      */
-    private function getMasterRequest(): Request
+    private function getMainRequest(): Request
     {
-        $masterRequest = $this->requestStack->getMasterRequest();
-        if (null === $masterRequest) {
+        if (\method_exists($this->requestStack, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ChannelBundle/Context/FakeChannel/FakeChannelContext.php:63]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists($this->requestStack, 'getMainRequest')) {
```

File: src/Sylius/Bundle/ChannelBundle/spec/Context/FakeChannel/FakeChannelContextSpec.php [hunk 9]
```diff
@@ -45,7 +45,11 @@ function it_returns_a_fake_channel_with_the_given_code(
         Request $request,
         ChannelInterface $channel
     ): void {
-        $requestStack->getMasterRequest()->willReturn($request);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ChannelBundle/spec/Context/FakeChannel/FakeChannelContextSpec.php:48]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Bundle/ChannelBundle/spec/Context/FakeChannel/FakeChannelContextSpec.php [hunk 10]
```diff
@@ -56,7 +60,11 @@ function it_returns_a_fake_channel_with_the_given_code(
 
     function it_throws_a_channel_not_found_exception_if_there_is_no_master_request(RequestStack $requestStack): void
     {
-        $requestStack->getMasterRequest()->willReturn(null);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ChannelBundle/spec/Context/FakeChannel/FakeChannelContextSpec.php:63]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Bundle/ChannelBundle/spec/Context/FakeChannel/FakeChannelContextSpec.php [hunk 11]
```diff
@@ -67,7 +75,11 @@ function it_throws_a_channel_not_found_exception_if_provided_code_was_null(
         RequestStack $requestStack,
         Request $request
     ): void {
-        $requestStack->getMasterRequest()->willReturn($request);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ChannelBundle/spec/Context/FakeChannel/FakeChannelContextSpec.php:78]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Bundle/ChannelBundle/spec/Context/FakeChannel/FakeChannelContextSpec.php [hunk 12]
```diff
@@ -82,7 +94,11 @@ function it_throws_a_channel_not_found_exception_if_channel_with_given_code_was_
...
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ChannelBundle/spec/Context/FakeChannel/FakeChannelContextSpec.php:97]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Bundle/CoreBundle/Checkout/CheckoutResolver.php [hunk 13]
```diff
@@ -47,7 +47,12 @@ public function __construct(
 
     public function onKernelRequest(RequestEvent $event): void
     {
-        if (!$event->isMasterRequest()) {
+        if (\method_exists($event, 'isMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Checkout/CheckoutResolver.php:50]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists($event, 'isMainRequest')) {
```

File: src/Sylius/Bundle/LocaleBundle/Context/RequestBasedLocaleContext.php [hunk 14]
```diff
@@ -32,9 +32,13 @@ public function __construct(RequestStack $requestStack, LocaleProviderInterface
 
     public function getLocaleCode(): string
     {
-        $request = $this->requestStack->getMasterRequest();
+        if (\method_exists($this->requestStack, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/LocaleBundle/Context/RequestBasedLocaleContext.php:35]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists($this->requestStack, 'getMainRequest')) {
```

File: src/Sylius/Bundle/LocaleBundle/spec/Context/RequestBasedLocaleContextSpec.php [hunk 15]
```diff
@@ -35,7 +35,11 @@ function it_is_a_locale_context(): void
 
     function it_throws_locale_not_found_exception_if_master_request_is_not_found(RequestStack $requestStack): void
     {
-        $requestStack->getMasterRequest()->willReturn(null);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/LocaleBundle/spec/Context/RequestBasedLocaleContextSpec.php:38]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Bundle/LocaleBundle/spec/Context/RequestBasedLocaleContextSpec.php [hunk 16]
```diff
@@ -44,7 +48,11 @@ function it_throws_locale_not_found_exception_if_master_request_does_not_have_lo
         RequestStack $requestStack,
         Request $request
     ): void {
-        $requestStack->getMasterRequest()->willReturn($request);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/LocaleBundle/spec/Context/RequestBasedLocaleContextSpec.php:51]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Bundle/LocaleBundle/spec/Context/RequestBasedLocaleContextSpec.php [hunk 17]
```diff
@@ -56,7 +64,11 @@ function it_throws_locale_not_found_exception_if_master_request_locale_code_is_n
         LocaleProviderInterface $localeProvider,
         Request $request
     ): void {
-        $requestStack->getMasterRequest()->willReturn($request);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/LocaleBundle/spec/Context/RequestBasedLocaleContextSpec.php:67]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Bundle/LocaleBundle/spec/Context/RequestBasedLocaleContextSpec.php [hunk 18]
```diff
@@ -70,7 +82,11 @@ function it_returns_master_request_locale_code(
...
```
Comment (lchrusciel) [inline src/Sylius/Bundle/LocaleBundle/spec/Context/RequestBasedLocaleContextSpec.php:85]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Bundle/ShopBundle/EventListener/NonChannelLocaleListener.php [hunk 19]
```diff
@@ -59,7 +59,12 @@ public function __construct(
      */
     public function restrictRequestLocale(RequestEvent $event): void
     {
-        if (!$event->isMasterRequest()) {
+        if (\method_exists($event, 'isMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ShopBundle/EventListener/NonChannelLocaleListener.php:62]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists($event, 'isMainRequest')) {
```

File: src/Sylius/Bundle/ShopBundle/EventListener/SessionCartSubscriber.php [hunk 20]
```diff
@@ -45,7 +45,12 @@ public static function getSubscribedEvents(): array
 
     public function onKernelResponse(ResponseEvent $event): void
     {
-        if (!$event->isMasterRequest()) {
+        if (\method_exists($event, 'isMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ShopBundle/EventListener/SessionCartSubscriber.php:48]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists($event, 'isMainRequest')) {
```

File: src/Sylius/Bundle/ShopBundle/spec/EventListener/NonChannelLocaleListenerSpec.php [hunk 21]
```diff
@@ -67,7 +67,11 @@ function it_does_nothing_if_its_not_a_master_request(
         FirewallMap $firewallMap,
         RequestEvent $event
     ): void {
-        $event->isMasterRequest()->willReturn(false);
+        if (\method_exists(RequestEvent::class, 'isMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ShopBundle/spec/EventListener/NonChannelLocaleListenerSpec.php:70]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestEvent::class, 'isMainRequest')) {
```

File: src/Sylius/Bundle/ShopBundle/spec/EventListener/NonChannelLocaleListenerSpec.php [hunk 22]
```diff
@@ -85,7 +89,11 @@ function it_does_nothing_if_request_is_behind_no_firewall(
         Request $request,
         RequestEvent $event
     ): void {
-        $event->isMasterRequest()->willReturn(true);
+        if (\method_exists(RequestEvent::class, 'isMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ShopBundle/spec/EventListener/NonChannelLocaleListenerSpec.php:92]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestEvent::class, 'isMainRequest')) {
```

File: src/Sylius/Bundle/ShopBundle/spec/EventListener/NonChannelLocaleListenerSpec.php [hunk 23]
```diff
@@ -103,7 +111,11 @@ function it_does_nothing_if_request_is_behind_a_firewall_not_stated_upon_creatin
...
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ShopBundle/spec/EventListener/NonChannelLocaleListenerSpec.php:114]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestEvent::class, 'isMainRequest')) {
```

File: src/Sylius/Bundle/ShopBundle/spec/EventListener/NonChannelLocaleListenerSpec.php [hunk 24]
```diff
@@ -123,7 +135,11 @@ function it_does_nothing_if_request_locale_is_present_in_the_provider(
...
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ShopBundle/spec/EventListener/NonChannelLocaleListenerSpec.php:138]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestEvent::class, 'isMainRequest')) {
```

File: src/Sylius/Bundle/ShopBundle/spec/EventListener/NonChannelLocaleListenerSpec.php [hunk 25]
```diff
@@ -145,7 +161,11 @@ function it_does_nothing_if_request_locale_is_not_present_in_provider_and_reques
...
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ShopBundle/spec/EventListener/NonChannelLocaleListenerSpec.php:164]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestEvent::class, 'isMainRequest')) {
```

File: src/Sylius/Bundle/ShopBundle/spec/EventListener/NonChannelLocaleListenerSpec.php [hunk 26]
```diff
@@ -175,7 +195,11 @@ function it_redirect_to_default_locale_if_request_locale_is_not_present_in_provi
         RequestEvent $event,
         Router $router
     ): void {
-        $event->isMasterRequest()->willReturn(true);
+        if (\method_exists(RequestEvent::class, 'isMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ShopBundle/spec/EventListener/NonChannelLocaleListenerSpec.php:198]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestEvent::class, 'isMainRequest')) {
```

File: src/Sylius/Component/Channel/Context/CachedPerRequestChannelContext.php [hunk 27]
```diff
@@ -38,7 +38,11 @@ public function __construct(ChannelContextInterface $decoratedChannelContext, Re
 
     public function getChannel(): ChannelInterface
     {
-        $objectIdentifier = $this->requestStack->getMasterRequest();
+        if (\method_exists($this->requestStack, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Component/Channel/Context/CachedPerRequestChannelContext.php:41]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists($this->requestStack, 'getMainRequest')) {
```

File: src/Sylius/Component/Channel/Context/RequestBased/ChannelContext.php [hunk 28]
```diff
@@ -49,14 +49,18 @@ private function getChannelForRequest(Request $request): ChannelInterface
         return $channel;
     }
 
-    private function getMasterRequest(): Request
+    private function getMainRequest(): Request
     {
-        $masterRequest = $this->requestStack->getMasterRequest();
-        if (null === $masterRequest) {
+        if (\method_exists($this->requestStack, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Component/Channel/Context/RequestBased/ChannelContext.php:54]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists($this->requestStack, 'getMainRequest')) {
```

File: src/Sylius/Component/Channel/spec/Context/CachedPerRequestChannelContextSpec.php [hunk 29]
```diff
@@ -38,7 +38,11 @@ function it_caches_channels_for_the_same_request(
         Request $request,
         ChannelInterface $channel
     ): void {
-        $requestStack->getMasterRequest()->willReturn($request, $request);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Component/Channel/spec/Context/CachedPerRequestChannelContextSpec.php:41]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Component/Channel/spec/Context/CachedPerRequestChannelContextSpec.php [hunk 30]
```diff
@@ -54,7 +58,11 @@ function it_does_not_cache_channels_for_different_requests(
         ChannelInterface $firstChannel,
         ChannelInterface $secondChannel
     ): void {
-        $requestStack->getMasterRequest()->willReturn($firstRequest, $secondRequest);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Component/Channel/spec/Context/CachedPerRequestChannelContextSpec.php:61]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Component/Channel/spec/Context/CachedPerRequestChannelContextSpec.php [hunk 31]
```diff
@@ -70,7 +78,11 @@ function it_caches_channels_for_the_same_request_even_if_there_are_other_request
         ChannelInterface $firstChannel,
         ChannelInterface $secondChannel
     ): void {
-        $requestStack->getMasterRequest()->willReturn($firstRequest, $secondRequest, $firstRequest);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Component/Channel/spec/Context/CachedPerRequestChannelContextSpec.php:81]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Component/Channel/spec/Context/CachedPerRequestChannelContextSpec.php [hunk 32]
```diff
@@ -85,7 +97,11 @@ function it_does_not_cache_results_while_there_are_no_master_requests(
         ChannelInterface $firstChannel,
         ChannelInterface $secondChannel
     ): void {
-        $requestStack->getMasterRequest()->willReturn(null, null);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Component/Channel/spec/Context/CachedPerRequestChannelContextSpec.php:100]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Component/Channel/spec/Context/CachedPerRequestChannelContextSpec.php [hunk 33]
```diff
@@ -98,7 +114,11 @@ function it_caches_channel_not_found_exceptions_for_the_same_request(
         RequestStack $requestStack,
         Request $request
     ): void {
-        $requestStack->getMasterRequest()->willReturn($request, $request);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Component/Channel/spec/Context/CachedPerRequestChannelContextSpec.php:117]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Component/Channel/spec/Context/CachedPerRequestChannelContextSpec.php [hunk 34]
```diff
@@ -111,7 +131,11 @@ function it_does_not_cache_channel_not_found_exceptions_for_null_master_requests
         RequestStack $requestStack,
         ChannelInterface $channel
     ): void {
-        $requestStack->getMasterRequest()->willReturn(null, null);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Component/Channel/spec/Context/CachedPerRequestChannelContextSpec.php:134]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Component/Channel/spec/Context/RequestBased/ChannelContextSpec.php [hunk 35]
```diff
@@ -39,7 +39,11 @@ function it_proxies_master_request_to_request_resolver(
         Request $masterRequest,
         ChannelInterface $channel
     ): void {
-        $requestStack->getMasterRequest()->willReturn($masterRequest);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Component/Channel/spec/Context/RequestBased/ChannelContextSpec.php:42]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Component/Channel/spec/Context/RequestBased/ChannelContextSpec.php [hunk 36]
```diff
@@ -51,7 +55,11 @@ function it_throws_a_channel_not_found_exception_if_request_resolver_returns_nul
         RequestStack $requestStack,
         Request $masterRequest
     ): void {
-        $requestStack->getMasterRequest()->willReturn($masterRequest);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Component/Channel/spec/Context/RequestBased/ChannelContextSpec.php:58]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

File: src/Sylius/Component/Channel/spec/Context/RequestBased/ChannelContextSpec.php [hunk 37]
```diff
@@ -61,7 +69,11 @@ function it_throws_a_channel_not_found_exception_if_request_resolver_returns_nul
     function it_throws_a_channel_not_found_exception_if_there_is_no_master_request(
         RequestStack $requestStack
     ): void {
-        $requestStack->getMasterRequest()->willReturn(null);
+        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```
Comment (lchrusciel) [inline src/Sylius/Component/Channel/spec/Context/RequestBased/ChannelContextSpec.php:72]:
```suggestion
        // TODO: Remove ACL layer after bump to Symfony >5.4
        if (\method_exists(RequestStack::class, 'getMainRequest')) {
```

**Review by lchrusciel — APPROVED** [review]
I meant, that we should place this comment above all occurrence of this ACL ifs. But thats a minor stuff. The plan will be as follows:
1. Merge it to 1.11, to get rid of deprecation message
2. Bump Symfony requirements to Sf ^5.4 on master
3. Remove this ACL from the master
At the end, trait would live with us too short, to consider it as separate package in my opinion.

**Comment by lchrusciel** [comment]
The base of this pull-request was changed, you need fetch and reset your local branch 
if you want to add new commits to this pull request. **Reset before you pull, else commits
may become messed-up.**

Unless you added new commits (to this branch) locally that you did not push yet, 
execute `git fetch origin && git reset "main-request-change"` to update your local branch.
 
Feel free to ask for assistance when you get stuck :+1:

**Comment by lchrusciel** [comment]
Thanks, Stephane! :1st_place_medal:

**Pull request merged**